orion/pkg/controllers/base_controller_test.go

```diff
@@ -45,19 +45,24 @@ var classService mockClassService
 
 // Fake classService that implements ClassService interface
 type mockClassService struct {
-	mockGetAll                    func() ([]domains.Class, error)
+	mockGetAll                    func(bool) ([]domains.Class, error)
+	mockGetUnpublished            func() ([]domains.Class, error)
 	mockGetByClassId              func(string) (domains.Class, error)
 	mockGetByProgramId            func(string) ([]domains.Class, error)
 	mockGetBySemesterId           func(string) ([]domains.Class, error)
 	mockGetByProgramAndSemesterId func(string, string) ([]domains.Class, error)
 	mockCreate                    func(domains.Class) error
 	mockUpdate                    func(string, domains.Class) error
 	mockDelete                    func(string) error
+	mockPublish                   func([]string) error
```

Comment:
Return list of errors

Reply:
I think this pull request is big enough already. I'll put this in another pull request.